Added guidance modifier to notice

agliga opened the pull request:
## Description
Added `page-notice--guidance` modifier and `page-notice--light` for guidance.

## Context
* Added modifier together with `page-notice` in order to override `border` property
* Added new component only to DS6
* I changed the content to always have padding and removed it from the icon in order to get the `page-notice--light` to have the padding as well (since it does not have an icon)
* I could have added a default background in guidance and not have `page-notice--light`, however I was worried about overriding styles and having to put a stronger selector on the other modules. 

__note__ Feel free to give me better description/example text for all the doc page for guidance. 

## References
#817 #452

## Screenshots
![image](https://user-images.githubusercontent.com/1755269/65268414-f24f2480-dacb-11e9-93b2-02d74daf875a.png)

ianmcburnie commented:
It would be nice if we can keep the API the same in DS4, otherwise we create another divergence in the docs. Let's think what `--guidance` might do to the DS4 version. Is there anything we can do that would look okay? Likewise with the `--light` modifier.

agliga commented:
At the moment adding`-guidance` on ds4 does nothing but does not break. It basically just renderes as a normal notice which isn't broken but looks the same. 
I can add light to ds4 but we need to worry about contrast. Maybe a very light gray around #eee
Otherwise if we want guidances to do something i can maybe add guidance to color all the borders not just top on ds4. 
Or we can just do the same colors and change on ds4

ianmcburnie commented:
@agliga @seangates I think guidance is supposed to remove visual weight, so what if we were to remove the thick top border in DS4? Or set that border to grey? As for light, let's just set it to very light grey background (maybe useful in the case of a white background area on a ds4 page). Realistically, I don't think anybody will use it anyway.

ianmcburnie commented:
@agliga I've just realised that a page-notice cannot have a light version. Only a guidance variant can. So what happens if I just have `page-notice page-notice--light`? It seems like a potential bad/unsupported combination, which we usually try to avoid.

ianmcburnie commented:
Actually, maybe disregard or rethink that last comment. I see that `--light` doesn't only set the background color, it affects layout too.

seangates commented on src/less/notice/ds6/notice.less:
```diff
@@ -13,6 +13,8 @@
 @page-notice-confirmation-background-color: #effef0;
 @page-notice-attention-border-color: @color-r4;
 @page-notice-attention-background-color: #fff5f8;
+@page-notice-light-background-color: #f7f7f7;
```
Are we 100% sure this doesn't fall under `@color-grey1: #f5f5f5;`, a color we already have?

agliga replied:
From what I spoke to design is they changed the background colors slightly in order to match color contrast ratios.
They said all the colors for the guidance/notice modules are different.

seangates replied:
I mean, it's so stinking close. 🤷‍♂

agliga commented:
> @agliga @seangates I think guidance is supposed to remove visual weight, so what if we were to remove the thick top border in DS4? Or set that border to grey? As for light, let's just set it to very light grey background (maybe useful in the case of a white background area on a ds4 page). Realistically, I don't think anybody will use it anyway. The idea is though that these modifiers make sense in any design system, they are semantic rather than visual (well okay, the `--light` one is a _bit_ visual. Perhaps that name should be changed.)

If we remove the top border then there will be no visual difference in the box to the success/info/attention. The DS4 top border is the part that gets visual modification, so removing that would make all guidance alerts look the same (since their background does not change)

seangates commented:
@agliga I think @ianmcburnie is saying the guidance module would rely only on the icon color, with a gray border.

agliga commented:
* Removed the `--light` color. Made default be light
* Added support for DS4. I had to nest styles differently. Basically `border-color` would be replaced by `border: none`. However, `border-top` would take precedence before the `border: none`. So I nested it so that the styles would override properly if `attention`, `information`, or `confirmation` modifier was applied. 
* Nested DS6 modifiers (basically gave it stronger selection), so that it could be applied to the base class since the DS4 requires a stronger selector. 

Here are screenshots of the new changes:
![image](https://user-images.githubusercontent.com/1755269/65284728-4b7c7f80-daef-11e9-9331-a7137a0ae9a1.png)

![image](https://user-images.githubusercontent.com/1755269/65284729-4dded980-daef-11e9-88db-3865673ce95f.png)

ianmcburnie commented:
In the examples, can you please separate the light/default versions from the confirm/attention/info examples.  Something like:

"For a light, default guidance module, please omit the status modifier."

agliga commented:
> In the examples, can you please separate the light/default versions from the confirm/attention/info examples. Something like:
> 
> "For a light, default guidance module, please omit the status modifier."

@ianmcburnie Done. I added the guidance blurb on the demo pages, but kept the guidance grouped together.

![image](https://user-images.githubusercontent.com/1755269/65354234-dcad2e00-dba3-11e9-8bcb-1a0fee79b6cc.png)

ianmcburnie commented:
@agliga just those pesky conflicts for you to fix, then we can merge.
